MapboxNavigation/NavigationMapView.swift

```diff
@@ -1335,3 +1353,96 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
+
+extension NavigationMapView {
+       
+    /**
+     Receives coordinates for searching the map for buildings. If buildings are found, they will be highlighted in 2d or 3d depending on the `in3D` value.
+     `in3D` defaults to true
+     `buildingHighlightingEnabled` must be set to true before this function will work.
+     */
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) {
+        highlightBuildings(Set(coordinates.compactMap({ buildingIdentifier(at: $0) })), in3D: extrudesBuildings, extrudeAll: false)
+    }
+    
+    /**
+     Removes the highlight from all buildings highlighted by `highlightBuildings(for coordinates:, in3D:)`.
+     */
+    public func unhighlightBuildings() {
+        guard let highlightedBuildingsLayer = style?.layer(withIdentifier: StyleLayerIdentifier.buildingExtrusion) else { return }
+        
+        style?.removeLayer(highlightedBuildingsLayer)
+    }
+    
+    private func addBuildingsLayer() -> MGLFillExtrusionStyleLayer? {
+        if let highlightedBuildingsLayer = style?.layer(withIdentifier: StyleLayerIdentifier.buildingExtrusion) as? MGLFillExtrusionStyleLayer { return highlightedBuildingsLayer }
+        guard let buildingsSource = style?.source(withIdentifier: "composite") else { return nil }
+        
+        let highlightedBuildingsLayer = MGLFillExtrusionStyleLayer(identifier: StyleLayerIdentifier.buildingExtrusion, source: buildingsSource)
+        highlightedBuildingsLayer.sourceLayerIdentifier = "building"
+        highlightedBuildingsLayer.fillExtrusionColor = NSExpression(forConstantValue: buildingDefaultColor)
+        highlightedBuildingsLayer.fillExtrusionOpacity = NSExpression(forConstantValue: 0.05)
+        highlightedBuildingsLayer.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+        highlightedBuildingsLayer.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+        
+        style?.addLayer(highlightedBuildingsLayer)
+        
+        return highlightedBuildingsLayer
+    }
+
+    private func buildingIdentifier(at coordinate: CLLocationCoordinate2D) -> Int64? {
+        let screenCoordinate = convert(coordinate, toPointTo: self)
+
+        // To increase a chance of selecting correct building identifier filter out
+        // features which do not contain appropriate attributes.
+        let features = visibleFeatures(at: screenCoordinate).filter {
+            let extrude = $0.attribute(forKey: "extrude") as? String
+            let type = $0.attribute(forKey: "type") as? String
+            return extrude == "true" && type == "building"
+        }
+        
+        if let feature = features.first, let identifier = feature.identifier as? Int64 {
+            return identifier
+        }
+        
+        return nil
+    }
+    
+    private func highlightBuildings(_ buildingIdentifiers: Set<Int64>, in3D: Bool = false, extrudeAll: Bool = false) {
```

Comment:
Building identifiers aren’t buildings, so this first argument should be labeled `withIdentifiers:`.